tasks/main.yml

```diff
@@ -4,9 +4,27 @@
 - name: Add tsuru repositories.
   apt_repository: repo='ppa:tsuru/ppa' update_cache=yes
 
-- name: Install packages.
+- name: Install base packages.
   apt: update_cache=true name="{{ item }}"
-  with_items: packages
+  with_items:
+    - curl
+    - python-software-properties
+
+- name: Install latest version of Gandalf Server
+  apt: name=gandalf-server
+  when: gandalf_server_version is not defined
+
+- name: Install a specific version of Gandalf Server
+  apt: name=gandalf-server={{ gandalf_server_version }}
+  when: gandalf_server_version is defined
+
+- name: Install latest version of Archive Server
+  apt: name=archive-server
+  when: archive_server_version is not defined
+
+- name: Install a specific version of Archive Server
+  apt: name=archive-server={{ archive_server_version }}
+  when: archive_server_version is defined
```

Comment:
I didn't perceive it to take that much longer, but that is purely subjective. It could possibly make the logic more complicated and difficult to follow but e.g. breaking out into python in the with_items e.g.

```{% if archive_server_version is defined %}archive-server={{ archive_server_version }}{% elif %}archive-server{% endif %} ```

What do you think?